src/main/java/com/amplitude/api/AmplitudeClient.java

```diff
@@ -2293,14 +2293,11 @@ public void run() {
                         }
                     }
                 });
-            } else if (stringResponse.equals("invalid_api_key")) {
+            } else if (response.code() == 400 && stringResponse.equals("invalid_api_key")) {
                 logger.e(TAG, "Invalid API key, make sure your API key is correct in initialize()");
-            } else if (stringResponse.equals("bad_checksum")) {
+            } else if (response.code() == 400 && stringResponse.equals("bad_checksum")) {
                 logger.w(TAG,
                         "Bad checksum, post request was mangled in transit, will attempt to reupload later");
-            } else if (stringResponse.equals("request_db_write_failed")) {
```

Comment:
why this is removed?

Reply:
This is from legacy code and not used anymore. Tagged you in offline discussion.